optaplanner-examples/src/test/java/org/optaplanner/examples/common/app/SolverPerformanceTest.java

```diff
@@ -94,65 +92,60 @@ public void setUp() {
+    private void assertScoreAndConstraintMatches(SolverFactory<Solution_> solverFactory, Solution_ bestSolution,
+            Score_ bestScoreLimit) {
         assertThat(bestSolution).isNotNull();
-        InnerScoreDirectorFactory<Solution_, Score_> scoreDirectorFactory =
-                (InnerScoreDirectorFactory<Solution_, Score_>) ((DefaultSolverFactory<Solution_>) solverFactory)
-                        .getScoreDirectorFactory();
-        Score_ bestScore = (Score_) scoreDirectorFactory.getSolutionDescriptor().getScore(bestSolution);
-        ScoreDefinition<Score_> scoreDefinition = scoreDirectorFactory.getScoreDefinition();
-        Score_ bestScoreLimit = scoreDefinition.parseScore(bestScoreLimitString);
-        assertThat(bestScore.compareTo(bestScoreLimit))
+        ScoreManager<Solution_, Score_> scoreManager = ScoreManager.create(solverFactory);
+        Score_ bestScore = scoreManager.updateScore(bestSolution);
+        assertThat(bestScore)
                 .as("The bestScore (" + bestScore + ") must be at least the bestScoreLimit (" + bestScoreLimit + ").")
-                .isGreaterThanOrEqualTo(0);
-
-        try (InnerScoreDirector<Solution_, Score_> scoreDirector = scoreDirectorFactory.buildScoreDirector()) {
-            scoreDirector.setWorkingSolution(bestSolution);
-            Score_ score = scoreDirector.calculateScore();
-            assertThat(bestScore).isEqualTo(score);
-            if (scoreDirector.isConstraintMatchEnabled()) {
-                Map<String, ConstraintMatchTotal<Score_>> constraintMatchTotals =
-                        scoreDirector.getConstraintMatchTotalMap();
-                assertThat(constraintMatchTotals).isNotNull();
-                assertThat(constraintMatchTotals.values().stream()
-                        .map(ConstraintMatchTotal::getScore)
-                        .reduce(Score::add)
-                        .orElse(scoreDefinition.getZeroScore())).isEqualTo(score);
-                assertThat(scoreDirector.getIndictmentMap()).isNotNull();
-            }
+                .isGreaterThanOrEqualTo(bestScoreLimit);
+
+        try {
+            ScoreExplanation<Solution_, Score_> scoreExplanation = scoreManager.explainScore(bestSolution);
+            Map<String, ConstraintMatchTotal<Score_>> constraintMatchTotals =
+                    scoreExplanation.getConstraintMatchTotalMap();
+            assertThat(constraintMatchTotals).isNotNull();
+            assertThat(constraintMatchTotals.values().stream()
+                    .map(ConstraintMatchTotal::getScore)
+                    .reduce(Score::add)
+                    .orElse(bestScore.zero()))
+                            .isEqualTo(scoreExplanation.getScore());
+            assertThat(scoreExplanation.getIndictmentMap()).isNotNull();
+        } catch (IllegalStateException ex) {
+            throw new IllegalStateException("Constraint match likely not supported.", ex);
```

Comment:
(nitpick, not important)

Do we need that catch that original IllegalStateException and wrap it?
The user is calling assertScoreAndConstraintMatches() and the call to ScoreManager.explainScore() has this javadoc:

```
     * @throws IllegalStateException when constraint matching is disabled or not supported by the underlying score
     *         calculator, such as {@link EasyScoreCalculator}.
     */
    ScoreExplanation<Solution_, Score_> explainScore(Solution_ solution);
```

I would expect that the IllegalStateException thrown by ScoreManager clearly says if Constraint matches are not supported - and there's little value in wrapping it here with "likely not supported". Wdyt?